### ISSUE TITLE
fileName include "[]" can't  be Ignored

### ISSUE THREAD

**Issue by uddeeqtkg** [libgit2/libgit2sharp#1862]
.gitignore inside the repo has:
\[i\].t
aa.json

Code
 var status=repository.RetrieveStatus(@"D:\repo\[i].t")     //status =NewinWorkdir
 var status2=repository.RetrieveStatus(@"D:\repoaa.json")  //status2 =Ignored

**Comment by ethomson** [comment]
What does git say about this status?  Have you tried escaping the `[` and `]` in your gitignore?  Those are globbing characters, see [the gitignore documentation](https://git-scm.com/docs/gitignore)

**Comment by uddeeqtkg** [comment]
$ git status    the file can be Ignored

**Comment by ethomson** [comment]
Can you post a _more detailed_ reproduction step then?  What you've posted doesn't make sense.  You're calling `RetrieveStatus` on `D:\repo[i].t` and `D:\repoaa.json`.  Those are two files in `D:\`.  Is your git repository working directory in `D:\`?  If so, a `.gitignore` with `aa.json` wouldn't match `repoaa.json`.

But if I have a `.gitignore` with:

```
[i].t
aa.json
```

and I have these two files in my working directory:

```
[i].t
aa.json
```

then git status says:

```
Untracked files:
  (use "git add <file>..." to include in what will be committed)
	[i].t
```

In other words, `[i].t` is not ignored, because `[` and `]` are special characters.  If I update my `.gitignore` to be:

```
\[i\].t
aa.json
```

then `[i].t` is ignored as it should be.

Whether this works or not in libgit2 or LibGit2Sharp, I did not try.  If you can provide a more detailed reproduction step then that would be a good next step to help you.

**Comment by uddeeqtkg** [comment]
D:\repoaaa has  files
```
 [i].t
 aa.json
 bbb.json
.gitignore
```
 If I update my .gitignore to be:
```
\[i\].t
aa.json
```
then use  git status :
```
D:\repoaaa>git  status
On branch master

No commits yet

Untracked files:
  (use "git add <file>..." to include in what will be committed)
        .gitignore
        bbb.json

nothing added to commit but untracked files present (use "git add" to track)
```
 [i].t  is ignored
but in  LibGit2Sharp
```
            var status1 = Repo.RetrieveStatus("[i].t");
            var status2 = Repo.RetrieveStatus("aa.json");
            Debug.Print($"[i].t->{status1}");
            Debug.Print($"aa.json->{ status2}");
            RepositoryStatus status = Repo.RetrieveStatus();
            Debug.Print($"--------Untracked Files-----------");
            foreach (StatusEntry statusEntry in status.Untracked)
            {
                Debug.Print(statusEntry.FilePath);
            }
```
output 
```
[i].t->NewInWorkdir
aa.json->Ignored
--------Untracked Files-----------
.gitignore
[i].t
bbb.json
```
 [i].t  is not ignored
